src/main/kotlin/com/sourcegraph/cody/agent/protocol/ContextFile.kt

```diff
@@ -1,7 +1,26 @@
 package com.sourcegraph.cody.agent.protocol
 
+import com.google.gson.JsonDeserializationContext
+import com.google.gson.JsonDeserializer
+import com.google.gson.JsonElement
+import java.lang.reflect.Type
+import java.net.URI
+
 data class ContextFile(
-    val fileName: String,
+    val uri: URI,
     val repoName: String?,
     val revision: String?,
 )
+
+val contextFileDeserializer =
```

Comment:
For some reason, deserialization did not work out of the box.

Reply:
IMO, we shouldn't do that in this way. Are these deserialization issues caused by the String->URI change or something else has been changed/introduced?